utils/EventRepository.js

```diff
@@ -226,14 +69,16 @@ class EventRepository {
     }
 
     // modify the url for event specific endpoints
-    const url = this.url.slice(-1) == "/" ? this.url.slice(0, -1) : this.url;
-    // add items to the queue
-    this.payloadQueue.addItem({
-      url: `${url}/v1/${type}`,
-      headers,
-      message,
-    });
+    const url = this.url.slice(-1) === "/" ? this.url.slice(0, -1) : this.url;
+    if (this.useBeacon) {
+      const targetUrl = `${url}/beacon/v1/batch`;
```

Comment:
Set the target URL during queue init and get rid of the `useBeacon` variable

Reply:
done